src/DataFrames.jl

```diff
@@ -92,6 +92,8 @@ export @~,
 ##
 ##############################################################################
 
+const displaysize(io::IO) = Compat.displaysize()
```

Comment:
Oh, and the problem with defining this even on 0.5 is that we won't use `io` even there. So you really need to keep the original `if`.